clients/src/test/java/org/apache/kafka/common/requests/OffsetCommitRequestTest.java

```diff
@@ -54,89 +68,204 @@ public class OffsetCommitRequestTest {
+                    "review the equality contract enforced here and add/remove fields accordingly.");
         }
+
+        OffsetCommitRequestData expected = expectedRequest.data();
+        OffsetCommitRequestData actual = actualRequest.data();
+
+        assertEquals(expectedRequest.version(), actualRequest.version());
+        assertEquals(expected.groupId(), actual.groupId(), "Group id mismatch");
+        assertEquals(expected.groupInstanceId(), actual.groupInstanceId(), "Group instance id mismatch");
+        assertEquals(expected.generationId(), actual.generationId(), "Generation id mismatch");
+        assertEquals(expected.memberId(), actual.memberId(), "Member id mismatch");
+        assertEquals(expected.retentionTimeMs(), actual.retentionTimeMs(), "Retention time mismatch");
+
+        Function<OffsetCommitRequestTopic, List<OffsetCommitRequestPartition>> partitionSelector =
+            OffsetCommitRequestTopic::partitions;
+
+        Function<OffsetCommitRequestTopic, NameAndId> topicClassifier =
+            topic -> new NameAndId(topic.name(), topic.topicId());
+
+        BiFunction<NameAndId, OffsetCommitRequestPartition, TopicIdPartition> partitionClassifier =
+            (nameAndId, p) -> new TopicIdPartition(nameAndId.id(), p.partitionIndex(), nameAndId.name());
+
+        Function<OffsetCommitRequestData, Map<TopicIdPartition, OffsetCommitRequestPartition>> partitioner =
+            request -> partition(request.topics(), partitionSelector, topicClassifier, partitionClassifier);
+
+        assertEquals(partitioner.apply(expected), partitioner.apply(actual));
+    }
+
+    /**
+     * Compares the two {@link OffsetCommitResponse} independently of the order in which the
+     * {@link OffsetCommitResponseTopic} and {@link OffsetCommitResponsePartition} are defined in the response.
+     */
+    public static void assertResponseEquals(OffsetCommitResponse expected, OffsetCommitResponse actual) {
+        assertEquals(expected.throttleTimeMs(), actual.throttleTimeMs());
+        assertEquals(expected.errorCounts(), actual.errorCounts());
+
+        Function<OffsetCommitResponseTopic, List<OffsetCommitResponsePartition>> partitionSelector =
+            OffsetCommitResponseTopic::partitions;
+
+        Function<OffsetCommitResponseTopic, NameAndId> topicClassifier =
+            topic -> new NameAndId(topic.name(), topic.topicId());
+
+        BiFunction<NameAndId, OffsetCommitResponsePartition, TopicIdPartition> partitionClassifier =
+            (nameAndId, p) -> new TopicIdPartition(nameAndId.id(), p.partitionIndex(), nameAndId.name());
+
+        Function<OffsetCommitResponse, Map<TopicIdPartition, OffsetCommitResponsePartition>> partitioner =
+            response -> partition(response.data().topics(), partitionSelector, topicClassifier, partitionClassifier);
+
+        assertEquals(partitioner.apply(expected), partitioner.apply(actual));
+    }
+
+    private static <T, P> Map<TopicIdPartition, P> partition(
```

Comment:
This seems to be a quite complicated way to group `OffsetCommitRequestPartition` or `OffsetCommitResponsePartition` by `TopicIdPartition`, no? I would just write two methods to do just this.